src/FetchProvider.ts

```diff
@@ -44,7 +45,14 @@ export default class FetchProvider implements Provider {
       headers: {
         'Content-Type': 'application/json',
       },
-    });
+    };
+
+    if (args.apiKey) {
+      requestObject.headers['Authorization'] = `Bearer ${args.apiKey}`;
+      requestObject.headers['X-Lib-Agent'] = getLibAgent();
```

Comment:
In the [Go PR](https://github.com/unstoppabledomains/resolution-go/pull/38#discussion_r1135754404), you named this header as `x-lib-client` instead of `X-Lib-Agent`. Let's pick one and make them consistent.